EPMRPP-84613 Playwright tags are parsed to the RP's attributes and attached to the tests (#133)
reportportal/agent-js-playwright: README.md

# @reportportal/agent-js-playwright

Agent to integrate Playwright with ReportPortal.

- More about [Playwright](https://playwright.dev/)
- More about [ReportPortal](http://reportportal.io/)

## Installation

Install the agent in your project:

```cmd
npm install --save-dev @reportportal/agent-js-playwright
```

## Configuration

**1.** Create `playwright.config.ts` or `*.config.js` file with reportportal configuration:

```typescript
import { PlaywrightTestConfig } from '@playwright/test';

const RPconfig = {
  apiKey: '00000000-0000-0000-0000-000000000000',
  endpoint: 'https://your.reportportal.server/api/v1',
  project: 'Your reportportal project name',
  launch: 'Your launch name',
  attributes: [
    {
      key: 'key',
      value: 'value',
    },
    {
      value: 'value',
    },
  ],
  description: 'Your launch description',
  grep: 'fast', // to run only tests with 'fast' tag in the title

>   grepInvert: 'fast', // to exclude tests with 'fast' tag in the title

Reviewer: Should be removed as it is a part of playwright configuration.

| extendTestDescriptionWithLastError          | Optional   | true      | If set to `true` the latest error log will be attached to the test case description.                                                                                                                                                                                                                                                                                                     |
| grep                                        | Optional   | Not set   | To run all the test cases contains a specific tag i.e. "@fast"                                                                                                                                                                                                                                                                                                                        |

> | grepInvert                                  | Optional   | Not set   | To run all the test cases not contains a specific tag i.e. "@fast"                                                                                                                                                                                                                                                                                                                    |

Reviewer: @abdelrhman-arnos as I see you removed the mention about `grep`s at all. Please add a short description about it to the `## Reporting` section.

> | grepInvert                                  | Optional   | Not set   | To run all the test cases not contains a specific tag i.e. "@fast"                                                                                                                                                                                                                                                                                                                    |

Author: I've created a new section for the grep options associated with an example of how can the user configure it. Please re-review the changes.

If you want to add a data to the suite, you must pass the suite name as the last parameter.

> ##### addAttributes

Reviewer: Please update this section with a small code example and mention that users can now specify test attributes in test titles.

});
```

> You can now add test attributes in test titles using `@tag` notation. This provides a succinct way to include metadata in test reports. Any `@tag` in a jest test title will be used as an attribute in ReportPortal, but won't appear in the final test title on ReportPortal.

Reviewer: ```suggestion
You can now add test attributes in test titles using `@tag` notation. This provides a succinct way to include metadata in test reports. Any `@tag` in a playwright test title will be used as an attribute in ReportPortal, but won't appear in the final test title on ReportPortal.
```

Example:

```javascript

> test('should have a tag at the beginning of the test title', () => {

Reviewer: I suppose that we need to provide an example of the feature mentioned above.
```suggestion
test('@tag should have a tag at the beginning of the test title', () => {
```

>   ReportingApi.addAttributes([{ key: 'title', value: 'tag' }]);

Reviewer: So please remove this method usage just to avoid mixing two approaches in this example.
```suggestion
```
